interfaces.go

```diff
@@ -1,46 +1,96 @@
 package dgraph
 
+type DependencyType string
+
+const (
+	// OrDependency means the dependencies are resolved when all `AND`s and a single `OR` dependency is resolved.
+	// In other words, one OR short-circuits all ORs, but not ANDs.
+	OrDependency DependencyType = "or"
+	// AndDependency means the dependency is required no matter what.
+	AndDependency DependencyType = "and"
+	// CompletionDependency means the dependency will resolve due to either resolution or failure.
+	CompletionDependency DependencyType = "completion"
+	// ObviatedDependency is for dependencies that no longer have an effect due to a prior resolution.
+	// For example, if one OR is resolved, all other OR dependencies are changed to ObviatedDependency.
+	ObviatedDependency DependencyType = "obviated-dependency"
+	// SoftDependency means the dependency does not wait for this dependency to resolve.
+	// The dependency may be unresolved at the time of resolution.
+	//SoftDependency DependencyType = "soft"
```

Comment:
Should this code be removed?  (We can add it again when it is actually needed, if that time comes.)

Reply:
Yeah, I can remove it until a need is found.